composer/loggers/logger_hparams.py

```diff
@@ -175,9 +176,17 @@ def get_flattened_dict(data: Dict[str, Any], _prefix: List[str] = []) -> Dict[st
                 )
             self.extra_init_params["config"].update(config)
 
-        name_suffix = f"Rank {dist.get_global_rank()}"
-        name = f"{self.name}_{name_suffix}" if self.name else name_suffix
-        group = self.name if (not self.group and self.rank_zero_only) else self.group
+        # If name=None, wandb.init(..) will automatically produce a unique run name
+        # But for multi-rank grouped runs, we want to provide a group name ahead of time to link the runs
+        # So let's explicitly default the run name here in a consistent way for single-rank and multi-rank
+        self.name = self.name if self.name else coolname.generate_slug(2)
```

Comment:
Does this work for multi-rank training if the user does not set a random seed? I forget how we do this in the trainer init.

Reply:
If we wanted to ensure a random name for each run (but the same random name across all ranks), then I think we'd need to move this logic into the `init()` event of the wandb callback. I think it would be possible to 1) capture the state of python's RNG, 2) re-seed python's RNG based on, say, the unix timestamp, 3) generate a coolname, 4) share this coolname with a ` dist.broadcast` , and 5) restore python's RNG state. Perhpas this is overkill, in which case I'm fine leaving this as-is.

Reply:
If the user does not set a random seed, I thought that Composer uses a default? Otherwise we wouldn't have random seed consistency across ranks, which seems like it could break dataset handling and/or model initialization...

Reply:
OK after reading through `trainer.py`, I think I didn't understand how random seeds are handled. Actually, I'm now more confused as to how the WandB ranks are getting the same random name if each device's seed is `seed += dist.get_global_rank()`. Debugging...

Reply:
The `WandbLogger` get constructed before the trainer seeding is done. The flow is usually:
```
logger = WandBlogger(). # coolname defined here (or through wandb_hparams.initialize_object())
trainer = Trainer(loggers=[logger]) # seed += rank happens here
```